sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -101,20 +101,19 @@ impl PgLTree {
     /// creates ltree from an iterator with checking labels
     pub fn from_iter<I, S>(labels: I) -> Result<Self, PgLTreeParseError>
     where
-        S: Into<String>,
+        String: From<S>,
```

Comment:
Since you're calling `parse()` on it, we don't need to copy it to an owned string. I would make this `S: AsRef<str>` and then do `label.as_ref().parse()` in the `for` loop.

Reply:
We need owned `String`s inside the `PgLTreeLabel` anyways, I changed `PgLTreeLabel::new` to take `S: Into<String>` and changed the call in here from `parse()?` to `PgLTreeLabel::new`.

This saves copies when the caller doesn't need to own the string but leaves `&str` valid!